Mathlib/Algebra/Polynomial/Splits.lean

```diff
@@ -139,6 +139,27 @@ theorem splits_X_pow (n : ℕ) : (X ^ n).Splits i :=
 theorem splits_id_iff_splits {f : K[X]} : (f.map i).Splits (RingHom.id L) ↔ f.Splits i := by
   rw [splits_map_iff, RingHom.id_comp]
 
+theorem Splits.comp_X_sub_C {i : L →+* F} (a : L) {f : L[X]}
+    (h : f.Splits i) : (f.comp (X - C a)).Splits i := by
+  cases h with
+  | inl h0 =>
+    left
+    simp only [map_eq_zero] at h0 ⊢
+    exact h0.symm ▸ zero_comp
+  | inr h =>
+    right
+    intro g irr dvd
+    rw [map_comp, Polynomial.map_sub, map_X, map_C, dvd_comp_X_sub_C_iff] at dvd
+    have := h (irr.map (algEquivAevalXAddC _)) dvd
+    change degree (g.comp (X + C (i a))) = 1 at this
```

Comment:
Why do you need `change` here? Please try to avoid it

Reply:
I found the correct theorem `Polynomial.comp_eq_aeval`.